client/packages/programs/src/JsonForms/components/IdGenerator.tsx

```diff
@@ -350,6 +344,11 @@ const UIComponent = (props: ControlProps) => {
   );
 
   const { text, onChange } = useDebouncedTextInput(data, manualUpdate);
+  const requireConfirmation = !options?.confirmRegenerate ? false : !!text;
```

Comment:
Actual fix as per Marks suggestion. The rest are just side quests. Makes reading the file a little bit easier. Tempted to refactor